portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -51,18 +56,25 @@ private Comment(String nameInput, Long timestampInput, String contentInput) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the max num comments from input
+    int maxNumComments = getMaxNumComments(request, 5);
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
```

Comment:
To go a little further into what we were discussing in our 1:1, getting a [PreparedQuery](http://cloud/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery) does _not_ grab all the results of that query immediately. The `prepare` [method](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/BaseDatastoreService.html#prepare) only "prepare"s it, it doesn't actually execute it.

That happens in the call to `results.asList`, so your `FetchOptions` object will only pull `maxNumComments` entries.

It might be worthwhile renaming the `results` variable to something like `preparedQuery` to help prevent confusion around this.

Reply:
That makes a lot more sense that PreparedQuery only prepares the query